### DISCUSSION TITLE
Supporting upload files and other needs

### DISCUSSION THREAD

**Discussion by artola** [graphql/graphql-http#12] [Q&A] [answered]
My Relay client is able to upload files using `multipart form` ... reading the implementation code I can not figure out a way to configure the client to achieve it. For example, this line sets the `content-type`:

https://github.com/enisdenjo/graphql-http/blob/01c45d8013d9c0fa8e43009dd816ae5e4c47f5b5/src/client.ts#L240

If the `...headers` is moved last, at least there is a way to override the default values, or if `headers` is a function, pass the defaults as an argument and use the return value.

But, going further ... should this library cover all the typical scenarios/needs out of the box? I mean file-upload or @defer/@stream.

@enisdenjo  what is your thought?

**Comment by enisdenjo** [comment]
Hey @artola, thanks for asking this! I've recently added a ["only GraphQL over HTTP" section](https://github.com/graphql/graphql-http#only-graphql-over-http) in the readme explaining what the aim of this library is.

In essence, this reference implementation strictly and exclusively follows the [GraphQL over HTTP spec](https://graphql.github.io/graphql-over-http/), no more no less. If you're in need for a full-featured server, I advise you consult the ["servers" section](https://github.com/graphql/graphql-http#servers) instead.

**Reply by artola** [reply]
Thanks @enisdenjo . As a member of the [ChilliCream](https://chillicream.com/) team if I need a sever it will be **HotChocolate** 🥇 

But I look from the client side, all your packages are _state of the art_ and I am trying to bring them altogether into a Relay client with all the capabilities of the GraphQL spec (and upcoming) around a middleware system to plug custom requirements.

I really appreciate and thank you for your effort on OSS, making it so much easier to get started with GraphQL.

**Comment by enisdenjo** [comment, marked as answer]
`graphql-http` supports custom request params parsers starting from [v1.20.0](https://github.com/graphql/graphql-http/releases/tag/v1.20.0). See more in #100.
